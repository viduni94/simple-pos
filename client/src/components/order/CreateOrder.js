import React, { Component } from "react";
import { connect } from "react-redux";
import PropTypes from "prop-types";

import Menu from "../menu/Menu";

export class CreateOrder extends Component {
  constructor(props) {
    super(props);

    let curr = new Date();
    curr.setDate(curr.getDate());

    this.state = {
      orderDate: curr,
      itemCount: "",
      status: "",
      userId: "",
      customerId: "",
      orderItems: "",
      errors: {}
    };
  }

  onChange = e => {
    this.setState({ [e.target.name]: e.target.value });
  };

  render() {
    return (
      <div className="create-order">
        <div className="row custom-row">
          <div className="col-md-7 ml-3">
            <Menu />
          </div>
          <div className="col-md-4 ml-5">
            <h2 className="text-center mu-title">New Order Details</h2>
            <div className="row col-md-12">
              <div className="col-md-6">
                <button className="btn btn-normal btn-block mr-4 mt-4 mb-4">Add New Customer</button>
              </div>
              <div className="col-md-6">
                <button className="btn btn-normal btn-block ml-4 mt-4 mb-4">Find Existing Customer</button>
              </div>
            </div>
            <small className="d-block pb-3">* = required fields</small>
            <form action="add-experience.html">
              <div className="form-group">
                <input type="text" className="form-control form-control-lg" value={this.state.customerId} name="customerId" onChange={this.onChange} required />
                <small className="form-text text-muted">Customer Id *</small>
              </div>
              <div className="form-group">
                <input type="text" className="form-control form-control-lg" value={this.state.orderDate.toISOString().substr(0, 10)} name="orderDate" required readOnly />
                <small className="form-text text-muted">Order Date *</small>
              </div>
              {/* <div className="form-group">
                <select className="form-control form-control-lg" name="status">
                  <option value="0">* Select Professional Status</option>
                  <option value="Developer">Developer</option>
                  <option value="Junior Developer">Junior Developer</option>
                  <option value="Senior Developer">Senior Developer</option>
                  <option value="Manager">Manager</option>
                  <option value="Student or Learning">Student or Learning</option>
                  <option value="Instructor">Instructor or Teacher</option>
                  <option value="Intern">Intern</option>
                  <option value="Other">Other</option>
                </select>
                <small className="form-text text-muted">Give us an idea of where you are at in your career</small>
              </div> */}
              <div className="form-group">
                <table className="table">
                  <thead className="thead-light">
                    <tr>
                      <th scope="col">#</th>
                      <th scope="col">Item</th>
                      <th scope="col">Quantity</th>
                      <th scope="col">Price</th>
                    </tr>
                  </thead>
                  <tbody>
                    <tr>
                      <th scope="row">1</th>
                      <td>Mark</td>
                      <td>Otto</td>
                      <td>@mdo</td>
                    </tr>
                    <tr>
                      <th scope="row">2</th>
                      <td>Jacob</td>
                      <td>Thornton</td>
                      <td>@fat</td>
                    </tr>
                    <tr>
                      <th scope="row">3</th>
                      <td>Larry</td>
                      <td>the Bird</td>
                      <td>@twitter</td>
                    </tr>
                  </tbody>
                </table>
              </div>
              <div className="row col-md-12">
                <div className="col-md-6">
                  <button type="submit" className="btn btn-secondary btn-block mr-4">
                    Cancel
                  </button>
                </div>
                <div className="col-md-6">
                  <button type="submit" className="btn btn-normal btn-block ml-4">
                    Submit
                  </button>
                </div>
              </div>
            </form>
          </div>
        </div>
      </div>
    );
  }
}

CreateOrder.propTpes = {
  order: PropTypes.object.isRequired,
  errors: PropTypes.object.isRequired
};

const mapStateToProps = state => ({
  order: state.order,
  errors: state.errors
});

export default connect(mapStateToProps)(CreateOrder);
