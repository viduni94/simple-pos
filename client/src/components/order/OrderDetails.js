import React, { Component } from "react";
import PropTypes from "prop-types";
import { connect } from "react-redux";
import Spinner from "../common/spinner";
import moment from "moment";
import { Link } from "react-router-dom";

import { setActiveOrder, getOpenOrderList } from "../../actions/orderListActions";
import { getFoodItems } from "../../actions/itemActions";
import { setActivePage } from "../../actions/pageActions";

import Menu from "../menu/Menu";

class OrderDetails extends Component {
  constructor(props) {
    super(props);

    this.state = {
      orderId: this.props.match.params.id
    };
  }

  componentDidMount() {
    this.props.setActiveOrder(this.state.orderId);
    this.props.getOpenOrderList();
    this.props.getFoodItems();
    this.props.setActivePage("orderDetails");
  }

  render() {
    const { orderLists, activeOrder, loading } = this.props.orderList;

    let activeOrderContent;
    let activeOrderDetails;
    let activeOrderItems;
    let totalItemCount;
    let totalBillAmount;

    if (orderLists === null || loading) {
      activeOrderContent = (
        <div className="m-auto">
          <Spinner />
        </div>
      );
    } else {
      if (Object.keys(orderLists).length > 0) {
        activeOrderDetails = orderLists.filter(order => !order._id.localeCompare(activeOrder));
        if (activeOrderDetails.length > 0) {
          activeOrderItems = activeOrderDetails[0].orderItems.map((item, index) => (
            <tr key={item._id}>
              <th scope="row">{index + 1}</th>
              <td>{item.foodItem.name}</td>
              <td className="text-center">{item.itemCount}</td>
              <td className="text-right pr-5">{(item.foodItem.unitPrice / 100).toFixed(2)}</td>
            </tr>
          ));

          totalItemCount = activeOrderDetails[0].orderItems.reduce((acc, currValue) => {
            return acc + currValue.itemCount;
          }, 0);

          totalBillAmount = activeOrderDetails[0].orderItems.reduce((acc, currValue) => {
            return acc + currValue.foodItem.unitPrice;
          }, 0);

          activeOrderContent = (
            <>
              <p className="lead text-center mb-4">
                Order Date - <b>{moment(activeOrderDetails[0].orderDate).format("Do MMM YYYY")}</b>
              </p>
              <form action="add-experience.html">
                <div className="form-group">
                  <input type="text" className="form-control form-control-lg" name="customerId" value={activeOrderDetails[0].customer.fname + " " + activeOrderDetails[0].customer.lname} required readOnly />
                  <small className="form-text text-muted">Customer Name</small>
                </div>
                <div className="form-group custom-table">
                  <table className="table">
                    <thead className="thead-light">
                      <tr>
                        <th scope="col">#</th>
                        <th scope="col">Item</th>
                        <th scope="col" className="text-center">
                          Quantity
                        </th>
                        <th scope="col" className=" text-center">
                          Price (LKR)
                        </th>
                      </tr>
                    </thead>
                    <tbody>
                      {activeOrderItems}
                      <tr>
                        <th scope="row">Total</th>
                        <td />
                        <td className="text-center">
                          <b>{totalItemCount}</b>
                        </td>
                        <td className="text-right pr-5">
                          <b>{(totalBillAmount / 100).toFixed(2)}</b>
                        </td>
                      </tr>
                    </tbody>
                  </table>
                </div>
                <div className="row col-md-12">
                  <div className="col-md-6">
                    <Link to={"/dashboard"} className="btn btn-secondary btn-block mr-4">
                      Cancel
                    </Link>
                  </div>
                  <div className="col-md-6">
                    <button type="submit" className="btn btn-normal btn-block ml-4">
                      Submit
                    </button>
                  </div>
                </div>
              </form>
            </>
          );
        }
      }
    }

    return (
      <div>
        <div className="create-order">
          <div className="row custom-row">
            <div className="col-md-7 ml-3">
              <Menu />
            </div>
            <div className="col-md-4 ml-5">
              <h2 className="text-center mu-title">Details of Order No:</h2>
              <h5 className="text-center"> {activeOrder}</h5>
              {activeOrderContent}
            </div>
          </div>
        </div>
      </div>
    );
  }
}

OrderDetails.propTypes = {
  orderList: PropTypes.object.isRequired,
  page: PropTypes.object.isRequired,
  setActiveOrder: PropTypes.func.isRequired,
  getOpenOrderList: PropTypes.func.isRequired,
  getFoodItems: PropTypes.func.isRequired,
  setActivePage: PropTypes.func.isRequired
};

const mapStateToProps = state => ({
  orderList: state.orderList,
  item: state.item,
  page: state.page
});

export default connect(
  mapStateToProps,
  { setActiveOrder, getOpenOrderList, getFoodItems, setActivePage }
)(OrderDetails);
