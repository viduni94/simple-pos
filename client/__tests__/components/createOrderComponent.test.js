import React from "react";
import { shallow } from "enzyme";
import { CreateOrder } from "../../src/components/order/CreateOrder";
import { wrap } from "module";

const props = {
  order: {},
  page: {},
  errors: {},
  customer: {},
  item: { items: [{id: "1", name: "abc", unitPrice: 400, itemCount: 1}]},
  auth: { user: { id: "1" } },
  resetActiveCustomer: jest.fn().mockReturnThis(),
  setActivePage: jest.fn().mockReturnThis()
};

describe("Create Order Component", () => {
  test("renders", done => {
    const wrapper = shallow(<CreateOrder {...props} />);
    expect(wrapper).toHaveLength(1);
    done();
  });

  test("should render initial layout", () => {
    // when
    const component = shallow(<CreateOrder {...props} />);
    // then
    expect(component.getElements()).toMatchSnapshot();
  });
});
