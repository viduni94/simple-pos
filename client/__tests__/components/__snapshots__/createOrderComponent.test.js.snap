// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Create Order Component should render initial layout 1`] = `
Array [
  <React.Fragment>
    <div
      className="create-order"
    >
      <div
        className="row custom-row"
      >
        <div
          className="col-md-7 ml-3"
        >
          <Connect(Menu)
            callbackFromParent={[Function]}
          />
        </div>
        <div
          className="col-md-4 ml-5"
        >
          <h2
            className="text-center mu-title"
          >
            Create New Order
          </h2>
          <div
            className="row col-md-12"
          >
            <div
              className="col-md-6"
            >
              <button
                className="btn btn-normal btn-block mr-4 mt-4 mb-4"
                onClick={[Function]}
              >
                Add New Customer
              </button>
            </div>
            <div
              className="col-md-6"
            >
              <button
                className="btn btn-normal btn-block ml-4 mt-4 mb-4"
                onClick={[Function]}
              >
                Find Existing Customer
              </button>
            </div>
          </div>
          <form
            onSubmit={[Function]}
          >
            <Alert
              closeAriaLabel="Close"
              color="success"
              fade={true}
              isOpen={false}
              tag="div"
              transition={
                Object {
                  "appear": true,
                  "baseClass": "fade",
                  "baseClassActive": "show",
                  "enter": true,
                  "exit": true,
                  "in": true,
                  "mountOnEnter": false,
                  "onEnter": [Function],
                  "onEntered": [Function],
                  "onEntering": [Function],
                  "onExit": [Function],
                  "onExited": [Function],
                  "onExiting": [Function],
                  "tag": "div",
                  "timeout": 150,
                  "unmountOnExit": true,
                }
              }
            >
              Order submitted succesfully!
            </Alert>
            <Alert
              closeAriaLabel="Close"
              color="danger"
              fade={true}
              isOpen={false}
              tag="div"
              transition={
                Object {
                  "appear": true,
                  "baseClass": "fade",
                  "baseClassActive": "show",
                  "enter": true,
                  "exit": true,
                  "in": true,
                  "mountOnEnter": false,
                  "onEnter": [Function],
                  "onEntered": [Function],
                  "onEntering": [Function],
                  "onExit": [Function],
                  "onExited": [Function],
                  "onExiting": [Function],
                  "tag": "div",
                  "timeout": 150,
                  "unmountOnExit": true,
                }
              }
            >
              An error occurred while submitting order!
            </Alert>
            <div
              className="form-group"
            >
              <input
                className="form-control form-control-lg"
                name="customerName"
                readOnly={true}
                required={true}
                type="text"
                value=""
              />
              <small
                className="form-text text-muted"
              >
                Customer Name *
              </small>
            </div>
            <div
              className="form-group"
            >
              <input
                className="form-control form-control-lg"
                name="orderDate"
                readOnly={true}
                required={true}
                type="text"
                value="2019-03-14"
              />
              <small
                className="form-text text-muted"
              >
                Order Date *
              </small>
            </div>
            <div
              className="form-group custom-table-1"
            >
              <table
                className="table"
              >
                <thead
                  className="thead-light"
                >
                  <tr>
                    <th
                      scope="col"
                    >
                      #
                    </th>
                    <th
                      scope="col"
                    >
                      Item
                    </th>
                    <th
                      scope="col"
                    >
                      Quantity
                    </th>
                    <th
                      scope="col"
                    >
                      Price
                    </th>
                    <th
                      scope="col"
                    >
                      Action
                    </th>
                  </tr>
                </thead>
                <tbody>
                  <tr>
                    <th
                      scope="row"
                    />
                    <td />
                    <td
                      className="text-center"
                    >
                      <b />
                    </td>
                    <td
                      className="text-right pr-5"
                    >
                      <b />
                    </td>
                    <td />
                  </tr>
                </tbody>
              </table>
            </div>
            <div
              className="row col-md-12"
            >
              <div
                className="col-md-6"
              >
                <Link
                  className="btn btn-secondary btn-block mr-4"
                  replace={false}
                  to="/dashboard"
                >
                  Cancel
                </Link>
              </div>
              <div
                className="col-md-6"
              >
                <button
                  className="btn btn-normal btn-block ml-4"
                  type="submit"
                >
                  Submit Order
                </button>
              </div>
            </div>
          </form>
        </div>
      </div>
    </div>
    <Connect(AddCustomerModal)
      toggle={
        Object {
          "backdrop": true,
          "modal": false,
          "toggle": [Function],
        }
      }
    />
    <Connect(FindCustomerModal)
      toggle={
        Object {
          "backdrop": true,
          "modal": false,
          "toggle": [Function],
        }
      }
    />
  </React.Fragment>,
]
`;
